src/dotnet/APIView/APIViewWeb/Managers/PullRequestManager.cs

```diff
@@ -450,15 +454,19 @@ private async Task AssertPullRequestCreatorPermission(PullRequestModel prModel)
             // White list bot accounts to create API reviews from PR automatically
             if (!_allowedListBotAccounts.Contains(prModel.Author))
             {
-                var orgs = await _githubClient.Organization.GetAllForUser(prModel.Author);
-                var orgNames = orgs.Select(o => o.Login);
-                var result = await _authorizationService.AuthorizeAsync(
-                    null,
-                    orgNames,
-                    new[] { PullRequestPermissionRequirement.Instance });
-                if (!result.Succeeded)
+                var isAuthorized = await _openSourceManager.IsAuthorizedUser(prModel.Author);
+                if (!isAuthorized)
                 {
-                    _telemetryClient.TrackTrace($"API change detection permission failed for user {prModel.Author}.");
+                    // Check if PR is assigned to a different user in case PR author is not authorized
+                    // This is possible is PR is contributed by external user but assigned to an internal user
+                    if (prModel.Assignee != null && prModel.Assignee != prModel.Author)
```

Comment:
Is using the assignee a workaround for allowing folks to generate even for external PRs? While I think this might be an OK workaround it definitely isn't intuitive and would require folks to add an assignee and then retrigger the pipeline to get an APIView.

Reply:
We can create a pipeline that can be triggered manually to generate API review for external users and make this pipeline available only in internal project.

Reply:
Yeah I think having an explicit pipeline we can trigger would be the best way to workaround any issues where we need to retry. Lets file an issue and put it on the backlog.

Reply:
That said I don't have an issue with you having this fallback here as well.